src/sql/workbench/services/objectExplorer/common/objectExplorerService.ts

```diff
@@ -130,6 +132,7 @@ export class ObjectExplorerService implements IObjectExplorerService {
 	public _serviceBrand: any;
 
 	private _disposables: IDisposable[] = [];
+	private _treeDisposables: IDisposable[] = [];
```

Comment:
>private _treeDisposables: IDisposable[] = []; [](start = 1, length = 45)

Any particular reason you're making a separate array? Currently there's just the one tree so this seems kind of pointless. (I could foresee us adding more in the future - but until that happens no reason to add extra stuff we don't need IMO)

Reply:
Yes, because this._disposables is disposed when `objectExplorerService` gets disposed. However, in this case, we only want to dispose the 'serverTreeView`, without disposing the whole service itself. `this._disposables` might have other stuff from `objectExplorerService` that we don't want disposed.

Reply:
Yeah I wasn't suggesting adding it to _disposables. I'm saying we're already tracking the item itself with _serverTreeView so you can just delete this whole array itself.

(and as my other comment suggests I don't even think this should be the one doing the disposing)

---
In reply to: [276467491](https://github.com/Microsoft/azuredatastudio/pull/5076#discussion_r276467491) [](ancestors = 276467491)

Reply:
@Charles-Gagnon We need to dispose that array since the disposable function is coupled with OE service. I agree we should set it to empty though, updated PR.